api/v1alpha1/etcdcluster_webhook.go

```diff
@@ -42,9 +42,6 @@ var _ webhook.Defaulter = &EtcdCluster{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (r *EtcdCluster) Default() {
 	etcdclusterlog.Info("default", "name", r.Name)
-	if r.Spec.Replicas == 0 {
-		r.Spec.Replicas = 3
-	}
 	if r.Spec.Storage.Size.IsZero() {
 		r.Spec.Storage.Size = resource.MustParse("4Gi")
 	}
```

Comment:
`// +kubebuiler:default:="4Gi"` can be also added to type definition instead of webhook

Reply:
It is not possible for complex structs to set default value. The only way to define default value for `resource.Quantity` is by code.

I also checked your suggestion, it's not working, it remains zero.

Actually, we can leave it zero (without any defaults) and it'll be up to user to fill it

Reply:
Checked one more time using local Kind cluster. Value remains zero if I use marker. So, I think we have two options
- leave it zero by default
- fill it using defaulting webhook